sound/soc/soc-topology.c

```diff
@@ -1807,6 +1809,9 @@ static int soc_tplg_dai_create(struct soc_tplg *tplg,
 	ret = soc_tplg_dai_load(tplg, dai_drv, pcm, NULL);
 	if (ret < 0) {
 		dev_err(tplg->comp->dev, "ASoC: DAI loading failed\n");
+		kfree(dai_drv->playback.stream_name);
+		kfree(dai_drv->capture.stream_name);
+		kfree(dai_drv->name);
```

Comment:
yes, but it's not complete. If a couple of lines below snd_soc_register_dai() fails, don't we also get a memory leak?

Reply:
@plbossart The dai_drv will be added to tplg->comp->dobj_list before calling snd_soc_register_dai(). So, remove_dai() will free them even snd_soc_register_dai() fails.

Reply:
ok